akka-typed/src/main/scala/akka/typed/persistence/scaladsl/PersistentActor.scala

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.persistence.scaladsl
+
+import scala.collection.{ immutable ⇒ im }
+import akka.annotation.DoNotInherit
+import akka.annotation.InternalApi
+import akka.typed.Behavior.UntypedBehavior
+import akka.typed.Signal
+import akka.typed.persistence.internal.PersistentActorImpl
+import akka.typed.scaladsl.ActorContext
+
+object PersistentActor {
+  /**
+   * Create a `Behavior` for a persistent actor.
+   */
+  def immutable[Command, Event, State](
+    persistenceId: String,
+    initialState:  State,
+    actions:       Actions[Command, Event, State],
+    applyEvent:    (Event, State) ⇒ State): PersistentBehavior[Command, Event, State] =
+    persistentEntity(_ ⇒ persistenceId, initialState, actions, applyEvent)
+
+  /**
+   * Create a `Behavior` for a persistent actor in Cluster Sharding, when the persistenceId is not known
+   * until the actor is started and typically based on the entityId, which
+   * is the actor name.
+   *
+   * TODO This will not be needed when it can be wrapped in `Actor.deferred`.
+   */
+  def persistentEntity[Command, Event, State](
+    persistenceIdFromActorName: String ⇒ String,
+    initialState:               State,
+    actions:                    Actions[Command, Event, State],
+    applyEvent:                 (Event, State) ⇒ State): PersistentBehavior[Command, Event, State] =
+    new PersistentBehavior(persistenceIdFromActorName, initialState, actions, applyEvent,
+      recoveryCompleted = (_, state) ⇒ state)
+
+  sealed abstract class Effect[+Event, State]() {
```

Comment:
I kinda feel like there should be an escape hatch for setting the State directly.  The inverse of SideEffect.

Reply:
Thanks for looking into this! If you have part of the state persistent and managed via events, but also want part of the state to be modifiable directly from commands, our current thinking is that  using `Actor.deferred` and a `var` for the directly-modifiable state like in https://github.com/akka/akka/pull/23674/files#diff-be682d29ed550fd2c88b3ce6e8eeed26R297 might be a clean way to do it. What do you think?

Reply:
yes, changing the state without an event might be dangerous/surprising. If it's replayed (restarted) it will not have the same state, and that can matter when applying events that were persisted after "patching" the state.

Reply:
Actually, and sorry this might take a bit to explain, is as follows.  I will say I totally understand if you don't want to support this usecase because its a little convoluted. I also understand that while I have good behavior around this, it may encourage bad behavior.

I usually have multiple persists in response to a single command.  Right now I have a runtime (a shim over untyped), that basically handles flatMap'ing over multiple persists and getting the state immediately as if the event was persisted.  And I can build up some nice, sophisticated workflows with it.  So my little mini programs run and output `(S, Seq[E], ...)`.  So in this case it would be nice to have that exposed.  But I can work around it by just duplicating the effort.  But the case for escape hatching would be that I do understand how it can effect the recovery, but I have good hygiene so maybe an escape hatch would be nice.

Reply:
@raboof also yes, that mechanism is how I keep things like actorRef's (that are transient).  Thanks for the discussion.